source/renderer/app/api/api.js

```diff
@@ -655,6 +658,44 @@ export default class AdaApi {
     }
   };
 
+  restoreLegacyWallet = async (
+    request: RestoreLegacyWalletRequest
+  ): Promise<Wallet> => {
+    Logger.debug('AdaApi::restoreLegacyWallet called', {
+      parameters: filterLogData(request),
+    });
+    const { recoveryPhrase, walletName, spendingPassword } = request;
+    const walletInitData = {
+      name: walletName,
+      mnemonic_sentence: split(recoveryPhrase, ' '),
+      passphrase: spendingPassword || '',
+    };
+    try {
+      const wallet: LegacyAdaWallet = await restoreLegacyWallet(this.config, {
+        walletInitData,
+      });
+      Logger.debug('AdaApi::restoreLegacyWallet success', { wallet });
+      return _createWalletFromServerData(wallet);
```

Comment:
@daniloprates Please improve this. `_createWalletFromServerData` accepting AdaWallet which has different properties than LegacyAdaWallet. Also `_createWalletFromServerData` has property is `isLegacy` that is set to `false` and in this case should be `true`.

Reply:
Hey @tomothespian I have instead changed the `restoreLegacyWallet` method to add the missing keys, otherwise, I'd need to change the Wallet domain, and I think it's better to keep it like this for now. Of course if can be changed later, when we have more info. Could you check it again?